export_job/export_zip_from_manifest_job.py

```diff
@@ -0,0 +1,172 @@
+import asyncio
+from time import sleep
+import aiohttp
+import json
+import os
+import shutil
+from gen3.tools.download.manifest import (
+    describe_access_to_files_in_workspace_manifest,
+    download_files_in_workspace_manifest,
+    list_files_in_workspace_manifest,
+)
+from gen3.auth import Gen3Auth
+import requests
+import boto3
+import sys
+
+from temporary_api_key import TemporaryAPIKey
+
+
+CHUNK_SIZE = 10
+MANIFEST_FILENAME = "manifest.json"
+EXPORT_DIR = "export"
+OUTPUT_PREFIX = "[out] "
+MAX_DOWNLOAD_SIZE = 250000000
+
+
+async def build_manifest_from_study_ids(hostname, token, study_ids):
+    """
+    build a manifest from a list of metadata guids representing study ids
+    dumps result to gen3-sdk compatible manifest file
+    """
+    ongoing_requests = []
+    study_metadata = []
+
+    print(f"Assembling manifest for study ids: {study_ids}")
+
+    async with aiohttp.ClientSession() as client:
+        for study_id in study_ids:
+            ongoing_requests += [
+                client.get(
+                    f"https://{hostname}/mds/aggregate/metadata/guid/{study_id}",
+                    headers={"Authorization": f"bearer {token}"},
+                )
+            ]
+            if len(ongoing_requests) == CHUNK_SIZE:
+                study_metadata += await asyncio.gather(
+                    *[req.json() for req in await asyncio.gather(*ongoing_requests)]
+                )
+
+        study_metadata += await asyncio.gather(
+            *[req.json() for req in await asyncio.gather(*ongoing_requests)]
+        )
+
+    # print(f"Retrieved metadata: {study_metadata}")
+    manifest = []
+    for study in study_metadata:
+        if not study["__manifest"]:
+            print(
+                f"Study {study['project_number']} is missing __manifest entry. Skipping."
+            )
+        else:
+            manifest += study["__manifest"]
+
+    download_size = sum(file["file_size"] for file in manifest)
+    if download_size > MAX_DOWNLOAD_SIZE:
+        fail(f"Download size {download_size} exceeds 250MB limit.")
+
+    with open(MANIFEST_FILENAME, "w+") as f:
+        json.dump(manifest, f)
+
+
+def download_files(access_token, hostname):
+    """
+    download the files described by the local manifest and export them to a zip
+    """
+    with TemporaryAPIKey(token=access_token, hostname=hostname):
+        auth = Gen3Auth(refresh_file=TemporaryAPIKey.file_name)
+        list_files_in_workspace_manifest(MANIFEST_FILENAME)
+        describe_access_to_files_in_workspace_manifest(
+            hostname, auth, MANIFEST_FILENAME
+        )
+        download_files_in_workspace_manifest(
```

Comment:
FYI just making a note here (since the DRS download python SDK branch doesn't have a PR now): I think the way that SDK implement this currently is to gather all the access token from all related commons first, and then iterate through the manifest to download files.

There is a possiblilty that if the manifest is large or for whatever reasons some downloads took lonbger than 20 mins, then some access token will expire, resulting in failure in subsequent downloads.

Probably not a big deal for now becaus we have a relatively small file size limits

Reply:
Yes the pre-caching of WTS is a known issue and needs to be addressed. I will highlight this in the related feature doc.